include/circt/Dialect/RTL/RTLTypeDecls.td

```diff
@@ -0,0 +1,45 @@
+//===- RTLTypeDecls.td - RTL data type declaration ops and types ----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Support for type declarations in the RTL type system.
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// Declaration operations
+//===----------------------------------------------------------------------===//
+
+def TypedeclScopeOp : RTLOp<"typedecl_scope",
```

Comment:
Actually, I'd recommend renaming this scope to be less similar to `TypedeclOp`.  We can/will eventually support other type declarations (e.g. enums) and they will go in this scope.  I'd recommend naming this just `TypeScopeOp`